src/js/components/message/message-sent-view.jsx

```diff
@@ -0,0 +1,140 @@
+            datatable_loading         : false
+        };
+    }
+
+    componentDidMount() {
+        this.reloadDatatable();
+        this.datatable_reload_interval = setInterval(() => this.reloadDatatable(), 60000);
+    }
+
+    componentWillUnmount() {
+        clearTimeout(this.datatable_reload_interval);
+    }
+
+    reloadDatatable() {
+        this.setState({
+            datatable_loading: true
+        });
+
+        return API.listTransactionWithDataSent(this.props.wallet.address_key_identifier).then(data => {
+            const rows = [];
+            data.forEach((transaction, idx) => {
+                transaction?.transaction_output_attribute.forEach(attribute => {
+                    if (attribute?.value) {
+                        const outputAttributeValue = attribute.value;
+                        if (!outputAttributeValue.file_data) {
+                            return;
+                        }
+                        for (const fileHash of _.keys(outputAttributeValue.file_data)) {
+                            const message    = outputAttributeValue.file_data[fileHash];
+                            let empty_tx     = _.isNil(message.subject);
+                            message.subject  = empty_tx ? this.getPendingTxMessage() : message.subject;
+                            message.message  = empty_tx ? this.getPendingTxMessage() : message.message;
+                            const newRow     = {
+                                idx        : idx,
+                                date       : format.date(transaction.transaction_date),
+                                txid       : transaction.transaction_id,
+                                txid_parent: outputAttributeValue.parent_transaction_id,
+                                dns        : outputAttributeValue.dns,
+                                amount     : format.number(transaction.amount),
+                                subject    : message.subject,
+                                address    : transaction.address_to,
+                                message    : message.message,
+                                sent       : true
+                            };
+                            newRow['action'] = <>
+                                <DatatableActionButtonView
+                                    disabled={empty_tx}
+                                    history_path={'/message-view/' + encodeURIComponent(transaction.transaction_id)}
+                                    history_state={{...newRow}}
+                                    icon={'eye'}/>
+                                <DatatableActionButtonView
+                                    history_path={'/message-compose'}
+                                    history_state={{}}
+                                    icon={'envelope'}/>
+                            </>;
+                            rows.push(newRow);
+                            idx++;
+                            break;
```

Comment:
it doesn't make sense to make a loop just to break it on first run. pop first item from the array and process it